client/app/user/profile/profile.controller.js

```diff
@@ -9,23 +9,117 @@ angular.module('observatory3App')
+                    if (!submission.verified){
+                        u.unverified +=1;
+                    }
+                    else if (submission.smallgroup && !submission.bonus){
+                        u.small.normal +=1;
+                    }
+                    else if (submission.smallgroup){
+                        u.small.bonus +=1;
+                    }
+                    else if(submission.bonus){
+                        u.full.bonus +=1;
+                    }
+                    else{
+                        u.full.normal +=1;
+                    }
+                }
+                u.full.all = user.totalDates.length;
+                u.small.all = user.totalSmallDates.length;
+
+                u.full.greyWidth = 0;
+                u.small.greyWidth = 0;
+
+                if (u.full.all === 0){
+                    u.full.goodWidth = 0;
+                    u.full.greyWidth = 100;
+                }
+                else{
+                    if (u.full.normal + u.full.bonus >= u.full.all){
+                        u.full.goodWidth = 100;
+                    }
+                    else{
+                        u.full.goodWidth = 100 * (u.full.normal + u.full.bonus) / u.full.all ;
+                    }
+                }
+
+                if (u.small.all === 0){
+                    u.small.goodWidth = 0;
+                    u.small.greyWidth = 100;
+                }
+                else{
+                    if (u.small.normal + u.small.bonus >= u.small.all){
+                        u.small.goodWidth = 100;
+                    }
+                    else{
+                        u.small.goodWidth = 100 * (u.small.normal + u.small.bonus) / u.small.all ;
+                    }
+                }
+
+                $http.get('/api/commits/user/' + user.githubProfile).success(function(commits){
+                    $scope.user.commits = commits;
+                    });
+                // get all users projects information
+                $scope.projects = [];
+                user.projects.forEach(function(projectId){
+                  $http.get("/api/projects/" + projectId).success(function(project){
+                    $scope.projects.push(project);
+
+                   $http.get('/api/commits/user/' + user.githubProfile).success(function(commits){
+                       $scope.user.commits = commits;
+                   });
+
+                  });
+                });
             });
-          });
+        }
+        else{
+            //Get non-private user information
+            $http.get('/api/users/' + $stateParams.id).success(function(user){
+                $scope.user = user;
+                if (!$scope.user.bio){
+                    $scope.user.bio = "An awesome RCOS developer!";
+                }
+                $scope.originalRole = user.role;
+
+                $http.get('/api/commits/user/' + user.githubProfile).success(function(commits){
```

Comment:
since the commits aren't integrated yet I feel like you shouldn't even worry about calling a commit endpoint yet for any of this code